devito/passes/clusters/misc.py

```diff
@@ -72,30 +73,135 @@ def callback(self, clusters, prefix):
         return lifted + processed
 
 
-@timed_pass()
-def fuse(clusters):
+class Fusion(Queue):
```

Comment:
Is making a class really necessary, seems to be "single use" class

Reply:
yes, because it's a subclass of `Queue` which is in practice an implicit-tree-visitor. `Fusion` is performing an implicit-tree visit over a list of Clusters